kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/VendorCreditMemoDocument.java

```diff
@@ -0,0 +1,116 @@
+            // was entered by the user or not.
+            
+            if ( !docAuth.isAuthorizedByTemplate(this, 
+                    KFSConstants.CoreModuleNamespaces.KFS, 
+                    KFSConstants.PermissionTemplate.EDIT_BANK_CODE.name, 
+                    GlobalVariables.getUserSession().getPrincipalId()  ) ) {
+                synchronizeBankCodeWithPaymentMethod();        
+            } else {
+                // ensure that the name is updated properly
+                refreshReferenceObject( BANK );
+            }
+        }        
+    }
+
+    public void synchronizeBankCodeWithPaymentMethod() {
+        Bank bank = getPaymentMethodGeneralLedgerPendingEntryService().getBankForPaymentMethod( getPaymentMethodCode() );
+        if ( bank != null ) {
+            setBankCode(bank.getBankCode());
+            setBank(bank);
+        } else {
+            // no bank code, no bank needed
+            setBankCode(null);
+            setBank(null);
+        }
+    }
+
+    @Override
+    public void customizeExplicitGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySourceDetail postable, GeneralLedgerPendingEntry explicitEntry) {
+        super.customizeExplicitGeneralLedgerPendingEntry(postable, explicitEntry);
+        // if the document is not processed using PDP, then the cash entries need to be created instead of liability
+        // so, switch the document type so the offset generation uses a cash offset object code
+        if ( !getPaymentMethodGeneralLedgerPendingEntryService().isPaymentMethodProcessedUsingPdp(getPaymentMethodCode())) {
+            explicitEntry.setFinancialDocumentTypeCode(DOCUMENT_TYPE_NON_CHECK);
+        }
+    }
+
+    public boolean generateDocumentGeneralLedgerPendingEntries(GeneralLedgerPendingEntrySequenceHelper sequenceHelper) {
+        if (getGeneralLedgerPendingEntries() == null || getGeneralLedgerPendingEntries().size() < 2) {
+            LOG.warn("No gl entries for accounting lines.");
+            return true;
+        }
+
+        getPaymentMethodGeneralLedgerPendingEntryService().generatePaymentMethodSpecificDocumentGeneralLedgerPendingEntries(
+                this,getPaymentMethodCode(),getBankCode(), KRADConstants.DOCUMENT_PROPERTY_NAME + AccountingDocumentRuleBaseConstants.ERROR_PATH.DOCUMENT_ERROR_PREFIX + PurapPropertyConstants.BANK_CODE, getGeneralLedgerPendingEntry(0), true, true, sequenceHelper);
+        
+        return true;
+    }
+    
+    
+    protected PaymentMethodGeneralLedgerPendingEntryService getPaymentMethodGeneralLedgerPendingEntryService() {
+        if ( paymentMethodGeneralLedgerPendingEntryService == null ) {
+            paymentMethodGeneralLedgerPendingEntryService = SpringContext.getBean(PaymentMethodGeneralLedgerPendingEntryService.class);
+        }
+        return paymentMethodGeneralLedgerPendingEntryService;
+    }
+    
+    public KualiDecimal getGrandPreTaxTotalExcludingDiscount() {
```

Comment:
Where did this method come from? I don't see it in either the KFS 6 or KFS 3 code for this document class.